lib/gateways.js

```diff
@@ -95,6 +96,13 @@ function disableHttpGatewayRedirect(button) {
   if (button) button.state(button, gui.toggleStateDisabled);
 }
 
+function pin(address) {
+  let IPFS_API_URI = 'http://' + prefs.customGatewayHost + ':5001/api/v0';
+  let uri = ioservice.newURI(IPFS_API_URI+'/pin/add?arg='+address, null, null);
+  tabs.open(uri.spec);
```

Comment:
I think opening this in a [Sidebar](https://developer.mozilla.org/en-US/Add-ons/SDK/High-Level_APIs/ui#Sidebar) would be less-intrusive for a user (as a quick UX fix). 

Ideally, we would want to parse the response and display some kind of [notification](https://developer.mozilla.org/en-US/Add-ons/SDK/High-Level_APIs/notifications) instead.